test/index-test.js

```diff
@@ -24,6 +36,16 @@ const verifyHold = f => {
   return Promise.all([p0, p1])
 }
 
+const createCollectSink = out => ({
```

Comment:
Is this recreating the functionality of `collect` above?  Is it possible to use `collect` instead?

Reply:
Not exactly. `collect` calls to `runEffects` which turns the stream to promise and _awaits_ its resolution on stream end which never happens for hot infinite source. `createCollectSink` exposes a disposable result of calling to `run` to provide more control on when it should be disposed (after the second run).

Reply:
Ah, I see.  It's intentionally creating a gap where no one is observing the hot stream.  That makes sense, thanks for explaining.